rules/proto_rules.build_defs

```diff
@@ -418,19 +418,26 @@ def _protoc_rule(name, srcs, deps, language, plugin, protoc_flags, root_dir, pre
         visibility = visibility,
     )
 
-def protoc_binary(name, version, hashes=None, deps=None, visibility=None):
+def protoc_binary(name, url:str|dict = '', version:str = '', hashes=None, deps=None, visibility=None):
     """Downloads a precompiled protoc binary.
 
     You will obviously need to choose a version that is available on Github - there aren't
     necessarily protoc downloads for every protobuf release.
 
     Args:
       name (str): Name of the rule
-      version (str): Version of protoc to download (e.g. '3.4.0').
+      url (str | dict): The URL used to download protoc. Can be a single string or a dictionary mapping
+                        HOSTOS-HOSTARCH to URLs i.e. linux-amd64: 'https://...'. Either provide url or version, but not both.
+      version (str): The version of protoc to download (e.g. '3.4.0'). Protoc will be downloaded from https://github.com/protocolbuffers/protobuf/releases/downaload/...
+                     and the rule will use the current platforms OS and ARCH setting. Either provide url or version,
+                     but not both.
       hashes (list): Hashes to verify the download against.
       deps (list): Any other dependencies
       visibility (list): Visibility of the rule.
     """
+    if url and version:
+        raise ParseError("Either version or url should be provided but not both")
```

Comment:
Raise is being deprecated in favour of `fail('msg here')`

Reply:
Fixed